### PR TITLE
Collect backtrace information for complete callstack

### PR DESCRIPTION
**Pull request by jklinkenberg**
**Current Situation:**
Currently only the rip and callsite that immeditately called malloc are recorded.
Lower level functions that allocate data which are invoked from different locations in the program can not be distiguished.

**Extension:**
This extension adds information to the existing dump containing addresses for the complete callstack for data objects / allocations while preserving existing information.
That enables to identify unique allocation paths within the program.

### CODE REVIEW

File: install_everything.sh [hunk 1]
```diff
@@ -41,16 +41,16 @@ else
...
+
+#libpfm
+PFM_ROOT=$ROOT_DIR/libpfm
+PFM_INSTALL_ROOT=$ROOT_INSTALL_DIR/libpfm
 
 if ! [ -d $PFM_INSTALL_ROOT ]; then
     echo "Installing libpfm..."
     rm -rf $PFM_ROOT
-    git clone https://git.code.sf.net/p/perfmon2/libpfm4 $PFM_ROOT || exit 1
+    git clone git://git.code.sf.net/u/jklinkenberg/perfmon2 $PFM_ROOT || exit 1
```
Comment (trahay) [inline install_everything.sh]: Why change the perfmon repo ? Are there changes in your repo that are needed and can't be merged into the "official" perfmon repo ?

File: install_everything.sh [hunk 2]
```diff
@@ -108,7 +108,7 @@ if ! [ -d $NUMAP_INSTALL_ROOT ]; then
     echo "Installing numap..."
     rm -rf $NUMAP_ROOT || exit 1
 
-    git clone https://github.com/numap-library/numap.git $NUMAP_ROOT || exit 1
+    git clone https://github.com/jklinkenberg/numap.git $NUMAP_ROOT || exit 1
```
Comment (trahay) [inline install_everything.sh]: Why change the Numap repo ? Are there changes in your repo that are needed and can't be merged into the "official" numap repo ?

File: install_everything.sh (hunk 1 above)
Comment (jklinkenberg) [reply install_everything.sh]: Hi, I actually reverted that change in a later commit as it is not necessary any more.
The tab "Files changed" should not list install_everything.sh any more.

Background:
We have Icelake system where PEBS was not working for loads due to some missing code in the kernel source for that architecture. After reporting the issue, Intel wrote a kernel fix that now already be published.
My temporary perform2 repo implemented a workaround for Icelake as the new kernel was not yet available.

If it is cleaner I can also create a PR not including changes for that file at all

File: install_everything.sh (hunk 2 above)
Comment (jklinkenberg) [reply install_everything.sh]: Also related to my other comment. Can be ignored.

**Comment by trahay** [comment]
Thanks for the pull request ! I merged it to numamma master branch (and updated the repo URLs in the installation script)

**Pull request merged**